lib/generator.js

```diff
@@ -131,6 +134,42 @@ class Generator {
         }
       });
     });
+
+    this.mapBaseUrlToFolderResolver = null;
+    if (mapBaseUrlToFolder && mapBaseUrlToFolder.url) {
+      this.mapBaseUrlToFolderResolver =  {
+        order: 1,
+        canRead (file) {
+          console.log(`my file resolver: ${file.url} ${file.extension}`);
+          return true;
+        },
+        read(file, callback, $refs) {
+          const baseUrl = mapBaseUrlToFolder.url;
+          const baseDir = mapBaseUrlToFolder.folder;
+
+          console.log('root dir %s',DEFAULT_TEMPLATES_DIR);
+
+          return new Promise(((resolve, reject) => {
+            let path = file.url;
+            console.log('converting %s ...', path);
+            path = path.replace(baseUrl,baseDir);
+            console.log('... to file %s', path);
+
+            try {
+              fs.readFile(path, (err, data) => {
+                if (err) {
+                  reject(`Error opening file "${path}"`);
+                } else {
+                  resolve(data);
+                }
+              });
+            } catch (err) {
+              reject(`Error opening file "${path}"`);
+            }
+          }));
+        }
+      };
+    };
```

Comment:
Could you move this logic to separate util function inside `Generator` class? Currently with your changes, the constructor is very complex. Also util function should be called here https://github.com/asyncapi/generator/blob/5406966b73da20f6e9db111e61ed0b4c32203437/lib/generator.js#L293 like

```js
parserOptions.resolve = {resolver: this.mapBaseUrlToFolderResolver()};
```

and util function should return resolver (object) with `order` etc fields.

Reply:
Thanks for the omments so far 👍 
I am still working on some tests and hope to finish the next days.